e2e/Dockerfile.jenkins

```diff
@@ -0,0 +1,54 @@
+FROM ubuntu:20.04
+
+# Setting bash as the default shell
+RUN rm /bin/sh && ln -s /bin/bash /bin/sh
+
+# Set directories we'll copy things from and to (respectively)
+ARG BASE_DIR
+ARG DOCKER_BASE_DIR
+
+# Install dependencies before we actually
+# switch into the unprivileged jenkins user.
+RUN apt-get -qq update
+RUN apt-get -y -qq install curl git
+
+# Installing XVFB and other GUI-related deps for E2E tests
+ARG DEBIAN_FRONTEND=noninteractive
+RUN apt-get -y -qq install xvfb \
+    libgtk-3-0 \
+    libxtst6 \
+    libnotify-dev \
+    libgconf-2-4 \
+    libnss3 \
+    libxss1 \
+    libasound2 \
+    libgbm-dev \
+    --no-install-recommends \
+  && rm -rf /var/lib/apt/lists/*
+
+# Adding a jenkins user so that we can install
+# and access everything within the user's own folder
+# We must also give the jenkins user permissions to
+# the $DOCKER_BASE_DIR where the synthetics app is
+# Note: 1167 is Jenkin's $UID
+RUN useradd -m -r -u 1167 jenkins
+RUN usermod -s /bin/bash jenkins
+
+# Copying repo directory
+ADD --chown=jenkins $BASE_DIR $DOCKER_BASE_DIR
+
+USER jenkins
+
+# Installing NVM
+ENV NVM_DIR /home/jenkins/.nvm
+RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash
```

Comment:
I'd appreciate some input on whether you think we should change this.

Given it runs with the Jenkins user's privileges I don't think it'll be a problem to pipe install into bash, but I'm also not a specialist in these types of things so I'd appreciate further guidance here on whether we need to change this in any way.

CC @justinkambic @vigneshshanmugam @v1v

Reply:
I am fine with this approach. Btw, Is it possible to build this image and push it to observability docker registry which has xvfb support instead of doing this on each PR? /cc @v1v Thoughts?

Reply:
> which has xvfb support instead of doing this on each PR? /cc @v1v Thoughts?

It's possible to make use of the `packer_cache.sh` script, this particular script is provided by the infra-automation team to cache the required dependencies in the CI Workers.

For such, it's required some changes:
1) Ask/Open a PR in infra to support this project for this particular caching feature
2) Create the `.ci/packer_cache.sh` script similar to -> https://github.com/elastic/apm-agent-rum-js/blob/master/.ci/packer_cache.sh
3) Support in the CI to build the docker image with the cached docker images, when possible. NOTE: if there are changes in the dockerfile then it will be required to be built!
4) Agree with the docker tag in the `packer_cache.sh` and the Jenkinsfile.